kroxylicious-operator/install/01.ClusterRole.kroxylicious-operator-filter-record-encryption.yaml

```diff
@@ -0,0 +1,27 @@
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+---
+# The access rules for the resources the Kroxylicious Operator consumes
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kroxylicious-operator-filter-record-encryption
```

Comment:
I like the split roles per Filter, it means we could generate a set of manifests per Filter and not have to amend/generate a combined role. This might make the process of adding in your own custom filter CRD easier. Generate manifests (with some future tooling), apply them, update the Operator configuration to be aware of the new Kind. Sounds better than trying to amend a combined role.